src/vs/workbench/contrib/chat/browser/chatPasteProviders.ts

```diff
@@ -87,9 +89,35 @@ export class PasteImageProvider implements DocumentPasteEditProvider {
 			return;
 		}
 
-		widget.attachmentModel.addContext(imageContext);
+		const customEdit = {
+			resource: model.uri,
+			variable: imageContext,
+			undo: () => {
+				widget = this.chatWidgetService.getWidgetByInputUri(model.uri);
+				if (!widget) {
+					throw new Error('No widget found for undo');
+				}
+				widget.attachmentModel.delete(imageContext.id);
+			},
+			redo: () => {
+				widget = this.chatWidgetService.getWidgetByInputUri(model.uri);
+				if (!widget) {
+					throw new Error('No widget found for redo');
+				}
+				widget.attachmentModel.addContext(imageContext);
+			},
+			metadata: { needsConfirmation: false, label: imageContext.name }
+		};
 
-		return;
+		return {
+			edits: [{
+				insertText: '', title: 'Empty Edit', kind: new HierarchicalKind(Mimes.text), handledMimeType: Mimes.text,
```

Comment:
Re-use the kind from above here. Also you'll want `handledMimeType` to be the type of image that you found